os/drivers/video/video.c

```diff
@@ -857,22 +880,32 @@ static int video_takepict_start(FAR video_upperhalf_t *priv, int32_t capture_num
 		flags = enter_critical_section();
 
 		next_video_state = estimate_next_video_state(priv, CAUSE_STILL_START);
-		change_video_state(priv, next_video_state);
+		ret = change_video_state(priv, next_video_state);
 
 		leave_critical_section(flags);
 
+		if(ret < 0) {
+			goto error;
+		}
+
 		dma_container = video_framebuff_get_dma_container(&priv->still_inf.bufinf);
 		if (dma_container) {
 			/* Start video stream DMA */
 
-			video_devops->set_buftype(priv->dev, V4L2_BUF_TYPE_STILL_CAPTURE);
-			video_devops->set_buf(priv->dev, dma_container->buf.m.userptr, dma_container->buf.length);
+			ret = video_devops->set_buftype(priv->dev, V4L2_BUF_TYPE_STILL_CAPTURE);
+			if(ret < 0) {
+				goto error;
+			}
+			ret = video_devops->set_buf(priv->dev, dma_container->buf.m.userptr, dma_container->buf.length);
+			if(ret < 0) {
+				goto error;
```

Comment:
This can be moved to function. it called 3times.

Reply:
Done , create new function named "video_set_stream_param"